.github/workflows/release-draft-binaries.yml

```diff
@@ -433,38 +431,16 @@ jobs:
         with:
           ref: ${{ github.event.inputs.release_branch }}
 
-      - name: Install Rust toolchain
-        run: |
-          # This will allow us update to rust version indicated in our rust-toolchain.toml file
-          rustup show
-          rustup target add ${{ matrix.job.target }}
-
-      - name: Install Cross
-        if: matrix.job.use-cross == true
-        run: cargo install --version 0.2.4 cross
-
-      - shell: bash
-        if: matrix.job.target == 'ubuntu-20.04'
+      - name: Install Nix
         run: |
-          set -x
-          use_cross_build=${{ matrix.job.use-cross }}
-          if [[ $use_cross_build != true ]]; then
-            sudo apt update
-            sudo apt -y --no-install-recommends install libclang-dev clang
-          fi
+          sh <(curl -L https://nixos.org/nix/install) --daemon
 
-      - name: Build NIFs
+      - name: Build NIFs (using Nix)
         run: |
           set -ex
 
-          if [[ '${{ matrix.job.use-cross }}' == 'true' ]]; then
-            # When building using cross use rust crypto (slower) since the cross image doesn't have an up-to-date libclang
-            cross build --target ${{ matrix.job.target }} -p ockam_rust_elixir_nifs --release --no-default-features -F rust-crypto
-            exit
-          fi
-
-          rustup target add ${{ matrix.job.target }}
-          cargo build --target ${{ matrix.job.target }} -p ockam_rust_elixir_nifs --release  --no-default-features -F aws-lc
+          nix-env -iA nixpkgs.rustup nixpkgs.cargo nixpkgs.nix
+          nix-shell --run "cargo build --target ${{ matrix.job.target }} -p ockam_rust_elixir_nifs --release"
```

Comment:
We need have different build format for binaries that need to be built with cross

```bash
if [[ '${{ matrix.job.use-cross }}' == 'true' ]]; then
  # When building using cross use rust crypto (slower) since the cross image doesn't have an up-to-date libclang
  cross build --target ${{ matrix.job.target }} -p ockam_rust_elixir_nifs --release --no-default-features -F rust-crypto
  exit
fi

nix develop ./tools/nix#rust --command "cargo build --target ${{ matrix.job.target }} -p ockam_rust_elixir_nifs --release  --no-default-features -F aws-lc"
```

Reply:
Done